kube-client/Cargo.toml

```diff
@@ -26,6 +26,7 @@ client = ["config", "__non_core", "hyper", "http-body", "tower", "tower-http", "
 jsonpatch = ["kube-core/jsonpatch"]
 admission = ["kube-core/admission"]
 config = ["__non_core", "pem", "dirs"]
+refresh-oidc = ["client", "form_urlencoded"]
```

Comment:
note that this doesn't actually save any dependencies because kube-core already pulls in form_urlencoded, but a feature might still make sense

Reply:
Same as above, this feature here is mainly to allow https dependencies stay optional

Reply:
Then this makes sense. I have a suggestion of how to expose it all the way up to kube in a way that looks nice to users:

```diff
diff --git a/examples/Cargo.toml b/examples/Cargo.toml
index 7eb1f45..0ab126a 100644
--- a/examples/Cargo.toml
+++ b/examples/Cargo.toml
@@ -14,11 +14,12 @@ license = "Apache-2.0"
 release = false
 
 [features]
-default = ["openssl-tls", "kubederive", "ws", "latest", "runtime"]
+default = ["openssl-tls", "kubederive", "ws", "latest", "runtime", "refresh"]
 kubederive = ["kube/derive"]
 openssl-tls = ["kube/client", "kube/openssl-tls"]
 rustls-tls = ["kube/client", "kube/rustls-tls"]
 runtime = ["kube/runtime", "kube/unstable-runtime"]
+refresh = ["kube/oauth", "kube/oidc"]
 ws = ["kube/ws"]
 latest = ["k8s-openapi/v1_26"]
 
diff --git a/kube-client/Cargo.toml b/kube-client/Cargo.toml
index 15916ff..035cb7c 100644
--- a/kube-client/Cargo.toml
+++ b/kube-client/Cargo.toml
@@ -21,12 +21,12 @@ rustls-tls = ["rustls", "rustls-pemfile", "hyper-rustls"]
 openssl-tls = ["openssl", "hyper-openssl"]
 ws = ["client", "tokio-tungstenite", "rand", "kube-core/ws", "tokio/macros"]
 oauth = ["client", "tame-oauth"]
+oidc = ["client", "form_urlencoded"]
 gzip = ["client", "tower-http/decompression-gzip"]
 client = ["config", "__non_core", "hyper", "http-body", "tower", "tower-http", "hyper-timeout", "pin-project", "chrono", "jsonpath_lib", "bytes", "futures", "tokio", "tokio-util", "either"]
 jsonpatch = ["kube-core/jsonpatch"]
 admission = ["kube-core/admission"]
 config = ["__non_core", "pem", "dirs"]
-refresh-oidc = ["client", "form_urlencoded"]
 
 # private feature sets; do not use
 __non_core = ["tracing", "serde_yaml", "base64"]
diff --git a/kube/Cargo.toml b/kube/Cargo.toml
index a7ff60a..485a32f 100644
--- a/kube/Cargo.toml
+++ b/kube/Cargo.toml
@@ -21,6 +21,7 @@ rustls-tls = ["kube-client/rustls-tls"]
 openssl-tls = ["kube-client/openssl-tls"]
 ws = ["kube-client/ws", "kube-core/ws"]
 oauth = ["kube-client/oauth"]
+oidc = ["kube-client/oidc"]
 gzip = ["kube-client/gzip"]
 client = ["kube-client/client", "config"]
 jsonpatch = ["kube-core/jsonpatch"]
```

renamed the features slightly so they fit in nicer with the existing ones (but it would need to be propagated into the src)

the example feature name (refresh) doesn't matter much i just want all of these default enabled in there so users can quick check against clusters, and i hope it makes sense with my second comment that aims to make the oidc feature __additive__ (i.e. enabling it doesn't prevent the other oidc mode).

Reply:
Looks good, added